server/src/test/java/org/opensearch/cluster/metadata/MetadataTests.java

```diff
@@ -1364,6 +1366,55 @@ public void testValidateDataStreamsForNullDataStreamMetadata() {
         }
     }
 
+    public void testMetadataBuildInvocations() {
```

Comment:
Should we also do some assertion on content of the metadata(old or new) after verifying the count of invocations of internal methods ?

Reply:
Makes sense. Added those validations. Let me know if you more additions/suggestions for test cases.